galley/pkg/config/analysis/local/analyze.go

```diff
@@ -89,28 +98,30 @@ func NewSourceAnalyzer(m *schema.Metadata, analyzer *analysis.CombinedAnalyzer,
 		kuberesource.DefaultExcludedResourceKinds(),
 		serviceDiscovery)
 
-	return &SourceAnalyzer{
+	sa := &SourceAnalyzer{
 		m:                    m,
-		sources:              make([]event.Source, 0),
+		sources:              make([]precedenceSourceInput, 0),
 		analyzer:             analyzer,
 		transformerProviders: transformerProviders,
 		namespace:            namespace,
+		istioNamespace:       istioNamespace,
 		kubeResources:        kubeResources,
 		collectionReporter:   cr,
 	}
+
+	sa.addMeshConfigSource(meshcfg.Default())
```

Comment:
It looks like the meshcfg is added in a few different places (default, in-cluster, file). Does analysis pick the most recently added source or is the content merged in some way?

Reply:
Most recently added, we don't attempt to merge. Added a comment clarifying this. 

This is consistent with how other resources are handled as well: we don't merge any k8s resources or CRs either. MeshConfig is a bit of a special case, though, so perhaps it might make sense to eventually support merging. I figured that change could wait for another day if so.

Reply:
Superseded sounds reasonable here.